doc/source/rllib-training.rst

```diff
@@ -520,6 +520,142 @@ Custom metrics can be accessed and visualized like any other training result:
+            - The log-likelihood of the exploration action.
+        """
+        pass
+
+It takes the model’s output, the action distribution class, the model itself,
+a timestep, and an ``explore`` switch and outputs a tuple of 1) action and
+2) log-likelihood.
+
+On the highest level, the ``Trainer.compute_action`` and ``Policy.compute_action(s)``
+methods have a boolean ``explore`` switch, which is passed into
+``Exploration.get_exploration_action``. If ``None``, the value of
+``Trainer.config[“explore”]`` is used.
+Hence ``config[“explore”]`` describes the default behavior of the policy and
+e.g. allows switching off any exploration easily for evaluation purposes
+(see :ref:`CustomEvaluation`).
+
+The following are example excerpts from different Trainers' configs
+(see rllib/agents/trainer.py) to setup different exploration behaviors:
+
+.. code-block:: python
+
+    # All of the following configs go into Trainer.config.
+
+    # 1) Switching *off* exploration by default.
+    # Behavior: Calling `compute_action(s)` without explicitly setting its `explore`
+    # param will result in no exploration.
+    # However, explicitly calling `compute_action(s)` with `explore=True` will
+    # still(!) result in exploration (per-call overrides default).
+    "explore": False,
+
+    # 2) Switching *on* exploration by default.
+    # Behavior: Calling `compute_action(s)` without explicitly setting its
+    # explore param will result in exploration.
+    # However, explicitly calling `compute_action(s)` with `explore=False`
+    # will result in no(!) exploration (per-call overrides default).
+    "explore": True,
+
+    # 3) Example exploration_config usages:
+    # a) DQN: see rllib/agents/dqn/dqn.py
+    "explore": True,
+    "exploration_config": {
+       "type": "EpsilonGreedy",  # <- Exploration sub-class by name or full path to module+class
+                                 # (e.g. “ray.rllib.utils.exploration.epsilon_greedy.EpsilonGreedy”)
+       # Parameters for the Exploration class' constructor:
+       "initial_epsilon": 1.0,
+       "final_epsilon": 0.02,
+       "epsilon_timesteps": 10000,  # Timesteps over which to anneal epsilon.
+    },
+
+    # b) DQN Soft-Q: In order to switch to Soft-Q exploration, do instead:
+    "explore": True,
+    "exploration_config": {
+       "type": "SoftQ",
+       # Parameters for the Exploration class' constructor:
+       "temperature": 1.0,
+    },
+
+    # c) PPO: see rllib/agents/ppo/ppo.py
+    # Behavior: The algo samples stochastically by default from the
+    # model-parameterized distribution. This is the global Trainer default
+    # setting defined in trainer.py and used by all PG-type algos.
+    "explore": True,
+    "exploration_config": {
+       "type": "StochasticSampling",
+    },
+
+The following table lists all existing Exploration sub-classes and where they
+are currently used (by default):
+
+.. image:: images/rllib-exploration-api-table.png
```

Comment:
This table is informative. Can it be shown right after introducing the exploration config, but before the large blocks of code for Exploration class?

Also, 
- should include a google drawings link as a comment (shared public, view only) so that this can be edited in the future.
- can you export the table as a SVG?

Reply:
Done.